src/os_auth/main-client.c

```diff
@@ -367,9 +368,17 @@ int main(int argc, char **argv)
     }
 
     if(sender_ip){
-        char opt_buf[256] = {0};
-        snprintf(opt_buf,254," IP:'%s'",sender_ip);
-        strncat(buf,opt_buf,254);
+		/* Check if this is strictly an IP address using a regex */
+		if (OS_Regex("^\\d+.\\d+.\\d+.\\d+$", sender_ip))
```

Comment:
This is a valid regex for an IP address. However, we should support CIDR addresses, for example: `192.168.1.0/24`.

I suggest using `OS_IsValidIP()`.